### PR TITLE
Add Feng Bash Scripts and Slurm Scripts for NERSC

### PR DESCRIPTION
**Pull request by smallfishabc**
## Motivation

To prepare the training dataset, PDBs may need to be downloaded/processed from the existing dataset. 

CoDNaS folder stores bash scripts to automatically obtain PDBs and calculate P(r) for **different conformations of the same protein.**

PDB70_NERSC folder stores slrum, bash and python scripts to clean downloaded PDB70 dataset and generate P(r) curve with NERSC.

## How to test the behavior?
```
There is a REAME file in the CoDNaS folder to explain the usage of the scripts.

The slurm file in the PDB70_NERSC folder can be submitted to NERSC with SLURM command sbatch to run the script. 

We need to download and copy all scripts in PDB70_NERSC folder to the root directory of PDB70 database together with PDBs. 

We then need to mkdir for pdb/saxs_r/sequence under the root directory and then submit the job.
```

## Checklist

We may need to combine the Python file under the PDB70_NERSC folder as a function of the metfish package.

### CODE REVIEW

File: PDB70_NERSC/PDBtoSeq.py
Comment (ajtritt) [inline PDB70_NERSC/PDBtoSeq.py:1]: Please add this to the `metfish` package, and make an entry point [here](https://github.com/lbl-cbg/metfish/blob/48207dedc6fd0c8890f793c3b77d292be0066d07/pyproject.toml#L40)

File: Feng_bash_scripts/CoDNaS_FOXS_RAW_single_conf/full_clean.sh
Comment (ajtritt) [inline Feng_bash_scripts/CoDNaS_FOXS_RAW_single_conf/full_clean.sh:1]: Can you add a comment at the top describing what this script does?

File: PDB70_NERSC/parallel_PDB70.slurm [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#SBATCH --ntasks-per-node=1
+#SBATCH --cpus-per-task=4
+#SBATCH --time=00:05:00
+#SBATCH --constraint=cpu
+#SBATCH --qos=shared
+#SBATCH --account=m3513
+
+module load python
+conda activate /pscratch/sd/l/lemonboy/PDB_prep
+parallel -j 4 bash fixpdb_pr_parallel.bash -f {} -o . ::: *.pdb
```
Comment (ajtritt) [inline PDB70_NERSC/parallel_PDB70.slurm]:
```suggestion
ls  *.pdb | parallel -j 4 bash fixpdb_pr_parallel.bash -f {} -o . :::
```

**Review by ajtritt — CHANGES_REQUESTED** [review]
Please rename `Feng_bash_scripts` to something more descriptive, and then move it and `PDB70_NERSC` into a new directory title `scripts`.

File: scripts/PDB70_NERSC/fixpdb_pr_parallel_v2.bash [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#mkdir -p ${pr_loc}
+#echo "created saxs P(r) storage folder at ${pr_loc}"
+
+if [ -f "$file" ]; then
+   echo ${file}
+   filename=$(basename "$file")
+   echo ${filename}
+   pdbfixer ${file} --output "${pdb_loc}/fixed_${filename}" --keep-heterogens=none --replace-nonstandard --add-residues --verbose
+   calc-pr "${pdb_loc}/fixed_${filename}" -o "${pr_loc}/${filename}.pr.csv"
+   python PDBtoSeq.py  -f "${pdb_loc}/fixed_${filename}" -o "${seq_loc}/${filename}.fasta"
```
Comment (ajtritt) [inline scripts/PDB70_NERSC/fixpdb_pr_parallel_v2.bash]:
```suggestion
   python PDBtoSeq.py  -f "${pdb_loc}/fixed_${filename}" -o "${seq_loc}/${filename}.fasta" 
   echo ${file} >> complete.txt
```

File: scripts/PDB70_NERSC/fixpdb_pr_parallel_v2.bash [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+pr_loc=${outdir}/saxs_r
+
+#mkdir -p ${pdb_loc}
+#echo "created PDB storage folder at ${pdb_loc}"
+#mkdir -p ${seq_loc}
+#echo "created sequence storage folder at ${seq_loc}"
+#mkdir -p ${pr_loc}
+#echo "created saxs P(r) storage folder at ${pr_loc}"
+
+if [ -f "$file" ]; then
```
Comment (ajtritt) [inline scripts/PDB70_NERSC/fixpdb_pr_parallel_v2.bash]:
```suggestion
if [ -f "$file" ]; then
       done=`grep -c ${file} complete.txt`
       if [ "$done" -ne "0" ] ; then
             return
       fi
```

**Comment by smallfishabc** [comment]
Hi, the slurm and python scripts are ready to review. 
The processed training/testing dataset is stored at /global/cfs/cdirs/m3513/metfish/PDB70_verB_fixed_data with a detailed readme file.
I will continue the data transfer from my scratch to this folder after the NERSC maintenance.

**Review by ajtritt — APPROVED** [review]

**Pull request merged**